tests/pipeline/test_statistical.py

```diff
@@ -802,3 +879,139 @@ def test_factor_regression_method_two_factors(self, regression_length):
                 columns=assets,
             )
             assert_frame_equal(output_result, expected_output_result)
+
+
+class VectorizedBetaTestCase(ZiplineTestCase):
+
+    def compare_with_empyrical(self, dependents, independent):
+        INFINITY = 1000000  # close enough
+        result = vectorized_beta(
+            dependents, independent, allowed_missing=INFINITY,
+        )
+        expected = np.array([
+            empyrical_beta(dependents[:, i].ravel(), independent.ravel())
+            for i in range(dependents.shape[1])
+        ])
+        assert_equal(result, expected, array_decimal=7)
+        return result
+
+    @parameter_space(seed=[1, 2, 3], __fail_fast=True)
+    def test_matches_empyrical_beta_aligned(self, seed):
+        rand = np.random.RandomState(seed)
+
+        true_betas = np.array([-0.5, 0.0, 0.5, 1.0, 1.5])
+        independent = as_column(np.linspace(-5., 5., 30))
+        noise = as_column(rand.uniform(-.1, .1, 30))
+        dependents = 1.0 + true_betas * independent + noise
+
+        result = self.compare_with_empyrical(dependents, independent)
+        self.assertTrue((np.abs(result - true_betas) < 0.01).all())
+
+    @parameter_space(seed=[1, 2, 3], __fail_fast=True)
+    def test_nan_handling_matches_empyrical(self, seed):
+        rand = np.random.RandomState(seed)
+
+        true_betas = np.array([-0.5, 0.0, 0.5, 1.0, 1.5])
+        independent = as_column(np.linspace(-5., 5., 30))
+        noise = as_column(rand.uniform(-.1, .1, 30))
+        dependents = 1.0 + true_betas * independent + noise
+
+        # Fill 20% of the input arrays with nans randomly.
+        dependents[rand.uniform(0, 1, dependents.shape) < 0.2] = nan
+        independent[rand.uniform(0, 1, independent.shape) < 0.2] = nan
+
+        # Sanity check that we actually inserted some nans.
+        self.assertTrue(np.count_nonzero(np.isnan(dependents)) > 0)
+        self.assertTrue(np.count_nonzero(np.isnan(independent)) > 0)
```

Comment:
It's a very small chance, but doesn't this mean this test could fail randomly?

Reply:
We're seeding the rng, so this is still deterministic.